### PR TITLE
[#11878] Add status and comments to AccountRequest

### PR DESCRIPTION
**Pull request by jayasting98**
Part of #11878 

**Outline of Solution**

 

Added status (and its type, `AccountRequestStatus`) and comments attributes to `AccountRequest`.

### CODE REVIEW

**Review by xenosf — COMMENTED** [review]
LGTM! Just one comment

File: src/main/java/teammates/storage/sqlentity/AccountRequest.java
Comment (jayasting98) [reply src/main/java/teammates/storage/sqlentity/AccountRequest.java:52]: I was also thinking this. However, I felt the singular, `comment`, didn't feel exactly right either. Also, I saw `teammates.storage.sqlentity.Student` also had the same plural `comments` attribute.

https://github.com/TEAMMATES/teammates/blob/234218979f55d3eb5689186ad737dcbdd22b3df8/src/main/java/teammates/storage/sqlentity/Student.java#L15-L23

Nevertheless, I can change this, but I think we will also have to change this in the tech design doc (I see it in some places there too), if you still insist.

Comment (xenosf) [reply src/main/java/teammates/storage/sqlentity/AccountRequest.java:52]: I agree that it is hard to find an alternative name that sounds right. (I was thinking something along the lines of `commentsText`, but that seems a bit clunky.)

There may be a bit of confusion as the words "comment"/"comments" is also used in the context of feedback response comments. In that context, the plural "comments" would almost always be referring to a collection.

However, seeing as that context is quite separate from this one, leaving the string's name as `comments` is probably OK too.

Comment (jayasting98) [reply src/main/java/teammates/storage/sqlentity/AccountRequest.java:52]: @xenosf I think for this one, we can consider merging for now to get the ball rolling first. We can rename it later, if necessary, before the ARF feature branch is merged

**Review by xenosf — APPROVED** [review]
LGTM

File: src/client/java/teammates/client/scripts/sql/DataMigrationForAccountRequestSql.java
Comment (jayasting98) [reply src/client/java/teammates/client/scripts/sql/DataMigrationForAccountRequestSql.java:47]: From my understanding, the feature branch will only be merged after the migration is completed, right? Is this no longer the case? Assuming this is still the case, the script should not be used anymore when this is merged so I was not too careful on this. This was mainly placed there to prevent compilation errors. 😅

Regardless, I did specifically choose `APPROVED` because it does in fact correspond to `registeredAt` being null. What you are describing, where `registeredAt` is non-null and an account has been created, actually corresponds to `REGISTERED`.

There was also a discussion on this in the issue. It was long ago though, so maybe you missed it. Just in case, see:
- https://github.com/TEAMMATES/teammates/issues/11878#issuecomment-1173783073
- https://github.com/TEAMMATES/teammates/issues/11878#issuecomment-1173803204
- https://github.com/TEAMMATES/teammates/issues/11878#issuecomment-1173816614

Of course, there is the case where it is set to `APPROVED` when it should actually in fact be `REGISTERED`, but I felt it was not a big problem because of the assumption above, and because the account was already created so the account request could then already be ignored.

Still, I can try to change this so that it is more accurate by making it such that it is `APPROVED` if `registeredAt` is null or `REGISTERED` otherwise, instead of being default `APPROVED` always. This way, it should be correct even if migration is done after the feature branch is merged. In order for the other PR that depends on this to be submitted, let's merge this first; I will make this change to the migration script in a separate PR. I hope this is okay, @EuniceSim142

**Pull request merged**
